muni_portal/core/serializers.py

```diff
@@ -16,3 +19,55 @@ def to_representation(self, value):
         repr_value = super(SerializerMethodNestedSerializer, self).to_representation(value)
         if repr_value is not None:
             return self.serializer(repr_value, **self.serializer_kwargs).data
+
+
+class ImageSerializerField(Field):
+    """A custom serializer for image field."""
+
+    def to_representation(self, value):
+        return {
+            "url": value.file.url,
+            "width": value.width,
+            "height": value.height,
+            "alt": value.title
+        }
+
+
+class RelatedPersonPageSerializer(Field):
+    """A custom serializer for related PersonPage."""
+
+    read_only = True
+    write_only = False
+
+    @staticmethod
+    def get_representation(value):
+        return {
+            "id": value.id,
+            "title": value.title,
+            "slug": value.slug,
+            "url": value.url,
+            "icon_classes": value.icon_classes if hasattr(value, "icon_classes") else None,
+            "profile_image": None,
+            "profile_image_thumbnail": None,
+            "job_title": None,
+        }
+
+    def to_representation(self, value):
+        result = self.get_representation(value)
+        if value.specific.profile_image:
+            result["profile_image"] = ImageSerializerField().to_representation(
+                value.specific.profile_image
+            )
+            result["profile_image_thumbnail"] = ImageRenditionField("max-100x100").to_representation(
+                value.specific.profile_image
+            )
+        if value.specific.job_title:
+            result["job_title"] = value.specific.job_title
```

Comment:
the field should always be present, but null when no value is available. That gives a consistent schema with a clear indication of missing data.

Reply:
Defaults are here
https://github.com/OpenUpSA/muni-portal-backend/blob/15ac0b0c1c73b7caaf8adb8ba98b59d07d58c2ac/muni_portal/core/serializers.py#L50

Reply:
oh sorry, you're right